server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -370,6 +371,17 @@ void doXContent(XContentBuilder builder) throws IOException {
             builder.endObject();
         }
 
+        public Query termQuery(Object value, MultiTermQuery.RewriteMethod method, QueryShardContext context) {
+            assert value instanceof String;
+            String strValue = (String) value;
+            if (strValue.length() >= minChars) {
+                return super.termQuery(value, context);
+            }
+            WildcardQuery query = new WildcardQuery(new Term(name(), value + "?"));
```

Comment:
I think it'd be safer to create an automaton manually and then instantiate an AutomatonQuery, otherwise there could be surprises if `value` contains `?` or `*`.